src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -164,11 +168,49 @@ QgsValueRelationFieldFormatter::ValueRelationCache QgsValueRelationFieldFormatte
 
 QStringList QgsValueRelationFieldFormatter::valueToStringList( const QVariant &value )
 {
+  // Note: the recommended way to pass a value is QVariant::StringList but
+  //       for back compatibility a string representation either in the form
```

Comment:
for backwards compatibility and also awesome formats like shp

Reply:
good point, should we add tests with shapefiles?

Reply:
there would certainly be nothing wrong about adding tests. Main point is: I don't want this code to look like "deprecated" because of a comment.